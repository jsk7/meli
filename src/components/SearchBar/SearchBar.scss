@import 'src/styles/variables';

.SearchBar {
  position: relative;

  input {
    width: 95%;
    height: 35px;
    border: 0;
    background-color: #fff;
    box-shadow:none;
    border-radius: 0px;
    padding: 0px $base-margin 0px 10px;
    border: 1px solid #ccc;
    vertical-align: middle;
    border-radius: 3px;
    font-size: $small-font;
    -webkit-appearance: none;
    outline: none;
  }

  .searchButton {
    border-radius: 0 3px 3px 0;
    width: 42px;
    height: 37px;
    right: 0px;
    border: 1px solid #ccc;
    background-color: #fff;
    cursor: pointer;
    position: absolute;
    display: inline-flex;
    background-image: linear-gradient(#fff,#f1f1f1);
  }
  .search {
    font-size: $small-font;
    position: relative;
    height: 30px;
    width: 30px;
    color: $dark;
    box-sizing: border-box;
    content: '\EA16';
    background-image: url('../../assets/ic_Search.png');
    background-position: center;
    background-repeat: no-repeat;
  }
}
