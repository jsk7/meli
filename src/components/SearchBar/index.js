import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router'
import './SearchBar.scss';

class SearchBar extends React.PureComponent {

  render() {
    const { value, onChange, search } = this.props;

    return (
      <div className="SearchBar">
        <form onSubmit={search}>
          <input type="text"
                  placeholder="Nunca dejes de buscar"
                  value={value}
                  onChange={onChange}
          />
          <button type="submit"
                  className="searchButton"
          >
            <i className="search" />
          </button>
        </form>
      </div>
    );
  }
}

SearchBar.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  search: PropTypes.func
}

export default withRouter(SearchBar);
