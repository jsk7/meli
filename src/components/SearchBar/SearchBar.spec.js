import React from 'react';
import SearchBar from './index';
import renderer from 'react-test-renderer';

describe('<SearchBar />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <SearchBar value={"textMock"} search={() => {}} onChange={() => {}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
});


