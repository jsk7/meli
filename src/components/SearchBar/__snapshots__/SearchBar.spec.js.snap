// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should render correctly 1`] = `
<div
  className="SearchBar"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Nunca dejes de buscar"
      type="text"
      value="textMock"
    />
    <button
      className="searchButton"
      type="submit"
    >
      <i
        className="search"
      />
    </button>
  </form>
</div>
`;
